src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/split_fc.cpp

```diff
@@ -0,0 +1,213 @@
+                for (int i = 0; i < split_num; ++i) {
+                    split_reshape_pattern_vec[i] = {fc_dim_vec[i], reshape_vec[1]};
+                }
+            }
+
+            std::vector<ov::Output<ov::Node>> zp_const_vec(split_num);
+            for (int i = 0; i < split_num; ++i) {
+                zp_const_vec[i] = need_to_split_convert ? split_cvts[i] : convert_node1_const->clone_with_new_inputs({});
+            }
+
+            for (int i = 0; i < split_num; ++i) {
+                auto sub_parent0 = std::make_shared<ov::op::v0::Convert>(split_wgts[i], cvt_prec);
+                auto sub_parent1 = std::make_shared<ov::op::v0::Convert>(zp_const_vec[i], cvt_prec);
+                ov::pass::disable_constant_folding(sub_parent0);
+                ov::pass::disable_constant_folding(sub_parent1);
+                auto sub_node = std::make_shared<ov::op::v1::Subtract>(sub_parent0, sub_parent1);
+
+                auto mul_node = std::make_shared<ov::op::v1::Multiply>(sub_node, split_muls[i]);
+                if (reshape_node) {
+                    auto reshape_pattern = ov::op::v0::Constant::create<int32_t>(ov::element::i32, ov::Shape{2}, split_reshape_pattern_vec[i]);
+                    wgt_node_vec[i] = std::make_shared<ov::op::v1::Reshape>(mul_node, reshape_pattern, reshape_node->get_special_zero());
+                } else {
+                    wgt_node_vec[i] = mul_node;
+                }
+            }
+        } else {
+            // get input
+            auto wgt_item = fc_node->get_input_node_shared_ptr(1);
+
+            // split weight
+            auto split_dim_range = wgt_item->get_shape()[split_dim];
+            const auto& wgt_shape = wgt_item->get_shape();
+
+            // needn't to split fc when the dim is 0.
+            if (split_dim_range <= 1 || ov::shape_size(wgt_shape) < 6600000) {
+                return false;
+            }
+
+            // We should use VariadicSplit to split input for FC.
+            auto fc_dim_vec = split_parts(split_dim_range, split_num);
+            auto split_length = ov::op::v0::Constant::create<int32_t>(ov::element::i32, ov::Shape{static_cast<size_t>(split_num)}, fc_dim_vec);
+            auto split_wgts = std::make_shared<ov::op::v1::VariadicSplit>(wgt_item,
+                                                                          split_dim_node,
+                                                                          split_length);
+
+            for (int i = 0; i < split_num; ++i) {
+                wgt_node_vec[i] = split_wgts->output(i);
+            }
+        }
+
+        // create fc Nodes according to the splited weight or splited pattern.
+        std::vector<std::shared_ptr<Node>> fc_node_vec(split_num);
+        for (int i = 0; i < split_num; ++i) {
+            fc_node_vec[i] = fc_node->clone_with_new_inputs(ov::OutputVector{src_item, wgt_node_vec[i]});
+            fc_node_vec[i]->get_rt_info()["parallelDomain"] = fc_node->get_friendly_name();
```

Comment:
In common case, it is not guaranteed that node friendly name is unique (despite the fact that we try to avoid duplications, it's "legal" from `ov::Model` perspective). Can it be a problem if several `FullyConnected` "numa groups" will have the same `parallelDomain` value? If yes, maybe it's better to store unique node name (we can get it via `get_name()` node method) in `parallelDomain`?

Reply:
updated.